src/main/java/nextstep/optional/User.java

```diff
@@ -33,7 +35,10 @@ public static boolean ageIsInRange1(User user) {
     }
 
     public static boolean ageIsInRange2(User user) {
-        return false;
+        return Optional.ofNullable(user)
+                .map(User::getAge)
+                .filter(age -> age != null && age >= 30 && age <= 45)
+                .isPresent();
```

Comment:
```suggestion
                .filter(Objects::nonNull)
                .filter(age -> age >= 30 && age <= 45)
                .isPresent();
```
Predicate 부분(`age -> age != null && age >= 30 && age <= 45`)을 메서드로 뺐다면 좋았을 것 같아요

또한 N번 더 반복하더라도 
가독성을 위해 filter 체인을 한번 더 사용하기도 합니다